src/rust/engine/process_execution/src/remote.rs

```diff
@@ -26,12 +26,79 @@ use super::{
 use std;
 use std::cmp::min;
 use workunit_store::{generate_random_64bit_string, get_parent_id, WorkUnit, WorkUnitStore};
+use std::fmt;
 
 // Environment variable which is exclusively used for cache key invalidation.
 // This may be not specified in an ExecuteProcessRequest, and may be populated only by the
 // CommandRunner.
 const CACHE_KEY_GEN_VERSION_ENV_VAR_NAME: &str = "PANTS_CACHE_KEY_GEN_VERSION";
 
+
+struct CancelRemoteExecutionToken {
+  // Remote execution process needs to be cancel in the case of Fatal error or timeout or
+  // pants run is cancelled by the user.
+  // CancelRemoteExecutionToken is used to cancel remote execution process by sending
+  // CancelOperationRequest.
+  operations_client: Arc<bazel_protos::operations_grpc::OperationsClient>,
+  cancel_op_req: bazel_protos::operations::CancelOperationRequest,
+  drop_enabled: bool,
```

Comment:
Can we call this something like `send_cancellation_on_drop` rather than `drop_enabled`? It's not clear to me what the behaviour is from the name right now :)